models/User.model.js

```diff
@@ -4,9 +4,13 @@ const { Schema, model } = require("mongoose");
 const userSchema = new Schema({
   username: {
     type: String,
+    required: [true, 'Username is required.'],
     unique: true
```

Comment:
only suggestion I would make is to add `trim:true` to the username field; just removes useless spaces